src/components/reviews/addReviewForm.jsx

```diff
@@ -29,15 +44,61 @@ class AddReviewForm extends React.Component {
   }
 
   changeCharacteristic(event) {
-    const characteristic = event.target.name;
+    const id = event.target.name;
     const value = event.target.value;
     this.setState({
-      characteristics: {...this.state.characteristics, [characteristic]: value}
+      characteristics: {...this.state.characteristics, [id]: value}
+    });
+  }
+
+  changeRating(num) {
+    this.setState({
+      rating: num
     });
   }
 
-  submit(event) {
+  validate(event) {
     event.preventDefault();
+    const errors = [];
+    // perform validation checks here
+    if (this.state.recommend === null) { // recommended radio buttons
+      errors.push(["Please choose whether to recommend this product.", 1]);
+    }
+    for (var charId in this.state.characteristics) { // characteristic radio buttons
+      if (this.state.characteristics[charId] === null) {
+        errors.push(["Please review all product characteristics.", 2]);
+        break;
+      }
+    }
+    if (!this.state.summary) { // summary
+      errors.push(["Please enter a review summary.", 3]);
+    }
+    if (this.state.body.length < 50) { // body
+      errors.push(["Please enter a review body of at least 50 characters.", 4]);
+    }
+    if (!this.state.nickname) { // nickname
+      errors.push(["Please enter your nickname.", 5]);
+    }
+    const emailFormat = /^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/;
```

Comment:
Lucky we both needed to do this around the same time, this would have taken me forever to figure out